service/header/service.go

```diff
@@ -19,6 +19,10 @@ type Service struct {
 	p2pServer     *P2PExchangeServer
 }
 
+// TODO @renaynay: how will we register core listener on the header Service? It's a part of header service but
+// 	we can't pass it directly to constructor b/c only Bridge nodes provide CoreListener, otherwise it'll always be nil
+// 	maybe we can make it an interface? Still hacky.
```

Comment:
This could be an issue describing the high level problem as well.

Reply:
> how will we register core listener on the header Service?

But should we? For now, we can only create and Start CoreListener without passing it anywhere, as you probably did. I also think that we can keep this the way it is rn unless we need to access CoreListener somehow. To me, it's like something hanging in the background without the need to be ever touched in the runtime, so putting it inside header service is not of an issue.

Reply:
```
// TODO @renaynay: how will we register core listener on the header Service? It's a part of header service but
// 	we can't pass it directly to constructor b/c only Bridge nodes provide CoreListener, otherwise it'll always be nil
// 	maybe we can make it an interface? Still hacky.

```

saving this comment here ^ 

---------------------------------------------------------------------------
I think the main issue is that all other header sub-services are passed to the header service and contained there while this sub-service is its own special case and is just delegated to the DI app to manage its lifecycle, and isn't tracked anywhere else.